core/pfe-core/controllers/internals-controller.ts

```diff
@@ -1,12 +1,85 @@
 import type { ReactiveController, ReactiveControllerHost } from 'lit';
 
-export class InternalsController implements ReactiveController {
+function isARIAMixinProp(key: string): key is keyof ARIAMixin {
+  return key === 'role' || key.startsWith('aria');
+}
+
+export class InternalsController implements ReactiveController, ARIAMixin {
+  declare role: ARIAMixin['role'];
+  declare ariaAtomic: ARIAMixin['ariaAtomic'];
+  declare ariaAutoComplete: ARIAMixin['ariaAutoComplete'];
+  declare ariaBusy: ARIAMixin['ariaBusy'];
+  declare ariaChecked: ARIAMixin['ariaChecked'];
+  declare ariaColCount: ARIAMixin['ariaColCount'];
+  declare ariaColIndex: ARIAMixin['ariaColIndex'];
+  declare ariaColIndexText: ARIAMixin['ariaColIndexText'];
+  declare ariaColSpan: ARIAMixin['ariaColSpan'];
+  declare ariaCurrent: ARIAMixin['ariaCurrent'];
+  declare ariaDisabled: ARIAMixin['ariaDisabled'];
+  declare ariaExpanded: ARIAMixin['ariaExpanded'];
+  declare ariaHasPopup: ARIAMixin['ariaHasPopup'];
+  declare ariaHidden: ARIAMixin['ariaHidden'];
+  declare ariaInvalid: ARIAMixin['ariaInvalid'];
+  declare ariaKeyShortcuts: ARIAMixin['ariaKeyShortcuts'];
+  declare ariaLabel: ARIAMixin['ariaLabel'];
+  declare ariaLevel: ARIAMixin['ariaLevel'];
+  declare ariaLive: ARIAMixin['ariaLive'];
+  declare ariaModal: ARIAMixin['ariaModal'];
+  declare ariaMultiLine: ARIAMixin['ariaMultiLine'];
+  declare ariaMultiSelectable: ARIAMixin['ariaMultiSelectable'];
+  declare ariaOrientation: ARIAMixin['ariaOrientation'];
+  declare ariaPlaceholder: ARIAMixin['ariaPlaceholder'];
+  declare ariaPosInSet: ARIAMixin['ariaPosInSet'];
+  declare ariaPressed: ARIAMixin['ariaPressed'];
+  declare ariaReadOnly: ARIAMixin['ariaReadOnly'];
+  declare ariaRequired: ARIAMixin['ariaRequired'];
+  declare ariaRoleDescription: ARIAMixin['ariaRoleDescription'];
+  declare ariaRowCount: ARIAMixin['ariaRowCount'];
+  declare ariaRowIndex: ARIAMixin['ariaRowIndex'];
+  declare ariaRowIndexText: ARIAMixin['ariaRowIndexText'];
+  declare ariaRowSpan: ARIAMixin['ariaRowSpan'];
+  declare ariaSelected: ARIAMixin['ariaSelected'];
+  declare ariaSetSize: ARIAMixin['ariaSetSize'];
+  declare ariaSort: ARIAMixin['ariaSort'];
+  declare ariaValueMax: ARIAMixin['ariaValueMax'];
+  declare ariaValueMin: ARIAMixin['ariaValueMin'];
+  declare ariaValueNow: ARIAMixin['ariaValueNow'];
+  declare ariaValueText: ARIAMixin['ariaValueText'];
+
   #internals: ElementInternals;
 
+  /** True when the control is disabled via it's containing fieldset element */
+  get formDisabled() {
+    return this.host.matches(':disabled');
```

Comment:
Will this trigger a re-render for the host element?

Reply:
No, see above comments. It's a platform limitation, you can't hijack the callback at runtime. Alternatives are a matching class decorator or a mixin.

Reply:
FYI this is what that would look like

```ts
import type { ReactiveElement } from 'lit';
import { InternalsController } from '../controllers/internals-controller.js';

function isReactiveElementClass(
  klass: Function // eslint-disable-line @typescript-eslint/ban-types
): klass is typeof ReactiveElement {
  return typeof klass['addInitializer' as keyof typeof klass] === 'function';
}

const instances = new WeakMap();

/**
 */
export function internals(instance?: ReactiveElement): InternalsController
export function internals(options?: Partial<ARIAMixin>): ClassDecorator
export function internals(optsOrInstance: unknown): InternalsController|ClassDecorator {
  if (optsOrInstance instanceof HTMLElement) {
    return instances.get(optsOrInstance);
  } else {
    return function(klass) {
      if (!isReactiveElementClass(klass)) {
        throw new Error(`@internals may only decorate ReactiveElements. ${klass.name} is does not implement ReactiveElement.`);
      }

      klass.addInitializer(instance => {
        const controller = new InternalsController(instance, optsOrInstance);
        instance.addController(controller);
        instances.set(instance, controller);
      });
    };
  }
}
```

```ts
@internals()
export class BaseButton extends LitElement {
  // ...
  #internals = internals(this);
  // ...
}
```

A mixin is arguably more honest here

Reply:
I see what you're saying.  So end users would be required to use a mixin over a controller.